core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -134,7 +136,8 @@ public static Endpoint unsafeCreate(String host, int port) {
 
     private static Endpoint create(String host, int port, boolean validateHost) {
         if (NetUtil.isValidIpV4Address(host)) {
-            return new Endpoint(host, host, port, DEFAULT_WEIGHT, HostType.IPv4_ONLY);
+            return new Endpoint(host, host, port, DEFAULT_WEIGHT, HostType.IPv4_ONLY,
+                                new DefaultAttributeMap());
```

Comment:
Metadata is an optional feature. Could lazily initiate it only when we really need it?